sql/references.sql

```diff
@@ -0,0 +1,32 @@
+-- Create a table giving each publication's references, identified by their merged_ids.
+-- The all_metadata_with_cld2_lid table contains references identified by vendor IDs for each orig_id.
+-- For each merged_id, we take all its orig_ids' references, and look up the merged_ids of the references.
+-- We exclude references that appear outside our merged corpus.
+SELECT
+  DISTINCT links1.merged_id AS merged_id,
+  links2.merged_id AS ref_id
+FROM (
+  SELECT
+    id,
+    reference
+  FROM
+    {{ staging_dataset }}.all_metadata_with_cld2_lid
+  CROSS JOIN
+    UNNEST(SPLIT(references, ",")) AS reference
+  WHERE
+    reference IN (
+    SELECT
+      orig_id
+    FROM
+      {{ staging_dataset }}.article_links_with_dataset )) AS references
+LEFT JOIN
+  {{ staging_dataset }}.article_links AS links1
```

Comment:
It'd be clearer to alias these joins something other than `links1` and `links2`. IIUC one gets the merged IDs for our referencing papers and the other gets the merged IDs for our referenced papers. How about calling them something like `referencing_papers` or `papers` and then `referenced_papers`?